rocAL/source/readers/image/caffe2_lmdb_record_reader.cpp

```diff
@@ -125,36 +165,53 @@ Reader::Status Caffe2LMDBRecordReader::folder_reading() {
     if (Caffe2_LMDB_reader() != Reader::Status::OK)
         WRN("Caffe2LMDBRecordReader ShardID [" + TOSTR(_shard_id) + "] Caffe2LMDBRecordReader cannot access the storage at " + _folder_path);
 
-    if (_in_batch_read_count > 0 && _in_batch_read_count < _batch_count) {
-        replicate_last_image_to_fill_last_shard();
-        LOG("Caffe2LMDBRecordReader ShardID [" + TOSTR(_shard_id) + "] Replicated " + _folder_path + _last_file_name + " " + TOSTR((_batch_count - _in_batch_read_count)) + " times to fill the last batch")
+        if (!_file_names.empty())
+        std::cout << "Caffe2LMDBRecordReader ShardID [" << TOSTR(_shard_id) << "] Total of " << TOSTR(_file_names.size()) << " images loaded from " << _full_path << std::endl;
+    auto dataset_size = _file_count_all_shards;
+    // Pad the _file_names with last element of the shard in the vector when _pad_last_batch_repeated is True
+    size_t padded_samples = 0;
+    // Pad the _file_names with last element of the shard in the vector when _pad_last_batch_repeated is True
+    padded_samples = ((_shard_size > 0) ? _shard_size : largest_shard_size_without_padding()) % _batch_size;
+    _last_batch_padded_size = (_batch_size > 1) ? (_batch_size - padded_samples) : 0;
+
+    if (_pad_last_batch_repeated == true) { 
+        // pad the last sample when the dataset_size is not divisible by
+        // the number of shard's (or) when the shard's size is not
+        // divisible by the batch size making each shard having equal
+        // number of samples
+        for (uint shard_id = 0; shard_id < _shard_count; shard_id++) {
+            uint start_idx = (dataset_size * shard_id) / _shard_count;
+            uint shard_size_without_padding = std::floor((shard_id + 1) * dataset_size / _shard_count) - floor(shard_id * dataset_size / _shard_count);
+            uint shard_size_with_padding = std::ceil(dataset_size * 1.0 / _shard_count);
+            auto start = _file_names.begin() + start_idx;
+            auto end = _file_names.begin() + start_idx + shard_size_without_padding;
+            auto start_file_size = std::next(_file_size.begin(), start_idx);
+            auto end_file_size = std::next(_file_size.begin(), start_idx + shard_size_without_padding);
+            if (start != end && start <= _file_names.end() &&
+                end <= _file_names.end()) {
+                _all_shard_file_names_padded.insert(_all_shard_file_names_padded.end(), start, end);
+                for (auto it = start_file_size; it != end_file_size; ++it) 
+                    _all_shard_file_sizes_padded.insert(*it);
+            }
```

Comment:
```suggestion
            if (start != end && start <= _file_names.end() &&
                end <= _file_names.end()) {
                _all_shard_file_names_padded.insert(_all_shard_file_names_padded.end(), start, end);
                for (auto it = start_file_size; it != end_file_size; ++it) 
                    _all_shard_file_sizes_padded.insert(*it);
            }
```

Check out this portion and see if it needs to be similar to file source reader, applicable for other readers too